cai_causal_graph/graph_components.py

```diff
@@ -442,6 +498,26 @@ def details(self) -> str:
         """Return a detailed string description of the object."""
         return self.__repr__()
 
+    @classmethod
+    def from_dict(cls, edge_dict: dict) -> Edge:
+        """
+        Deserialize a dictionary to a `cai_causal_graph.graph_components.Edge` instance.
+
+        :param edge_dict: The dictionary representation of the `cai_causal_graph.graph_components.Edge` instance.
+        :return: The `cai_causal_graph.graph_components.Edge` instance.
+        """
+        NodeCls = NodeClassDict[edge_dict['source']['node_class']]
```

Comment:
Why not check `destination` as well? Should we allow N -> TSN or require both to be same class?

Reply:
This will also raise a `KeyError` to user if the dict is bad which will be really confusing. Would maybe default to `Node` if class is not defined. So recommend the following (but I have a question if we want to force nodes to be same):
```
# Get node class for each node; defaulting to 'Node' if not defined in the dict.
source_node_class = edge_dict['source'].get('node_class', 'Node')
destination_node_class = edge_dict['source'].get('node_class', 'Node') 
assert source_node_class in NodeClassDict, 'NICE ERROR MESSAGE'
assert destination_node_class in NodeClassDict, 'NICE ERROR MESSAGE'
assert source_node_class == destination_node_class # DO WE WANT TO FORCE THIS?
NodeCls = NodeClassDict[source_node_class]
```

Reply:
I would cast them to the subclass.